### PR TITLE
Migrate Channel.GetChannelMembersTimezones to Sync by default

### LINKED ISSUE #11226: Migrate "Channel.GetChannelMembersTimezones" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16334)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetChannelMembersTimezones` in the `Channel` store.

The expected way to implement it is, go to the `Channel` store implementation in the `store/sqlstore/` directory, modify the method `GetChannelMembersTimezones` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by krjn**
Fixes #11226

https://mattermost.atlassian.net/browse/MM-16334

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1411,25 +1411,21 @@ func (s SqlChannelStore) GetMembers(channelId string, offset, limit int) store.S
...
 }
 
-func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		var dbMembersTimezone []map[string]string
-		_, err := s.GetReplica().Select(&dbMembersTimezone, `
+func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) ([]model.StringMap, *model.AppError) {
+	var dbMembersTimezone []model.StringMap
+	_, err := s.GetReplica().Select(&dbMembersTimezone, `
 					SELECT
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: nit: Do you mind to remove (at least 2 tabs) from each query lines?

**Comment by krjn** [comment]
> +1 LGTM. There were no tests to update? If test is missing for this method - please add one.

The function ends up being encapsulated and tested through `App.GetChannelMembersTimezones` on `app\channel.go`, which itself it tested . If requested I can create a specific tes method.

**Comment by jespino** [comment]
> I added another commit instead of force pushing as I'm not sure it'd be okay.

@krjn It is correct, and actually all the way around would be wrong. We prefer multiple commits and merges instead of rebases (because is easier for us to track the changes and do not review multiple times the same code). Besides, we always "Squash and merge" the PRs so it will end up being a single commit anyway.

**Comment by jespino** [comment]
Merged! Thanks @krjn! :fireworks:

**Pull request merged**